hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/HandleThrottleAccumulator.java

```diff
@@ -0,0 +1,491 @@
+        if (assetChange > 0) {
+            if (isSerializedProtoKey(alias)) {
+                return true;
+            } else {
+                return isOfEvmAddressSize(alias);
+            }
+        }
+
+        return false;
+    }
+
+    private boolean shouldThrottleBasedOnImplicitCreations(
+            final ThrottleReqsManager manager, final int implicitCreationsCount, final Instant now) {
+        return (implicitCreationsCount == 0)
+                ? !manager.allReqsMetAt(now)
+                : shouldThrottleImplicitCreations(implicitCreationsCount, now);
+    }
+
+    private boolean shouldThrottleImplicitCreations(final int n, final Instant now) {
+        final var manager = functionReqs.get(CRYPTO_CREATE);
+        return manager == null || !manager.allReqsMetAt(now, n, ONE_TO_ONE);
+    }
+
+    public void rebuildFor(@NonNull ThrottleDefinitions defs) {
+        List<DeterministicThrottle> newActiveThrottles = new ArrayList<>();
+        EnumMap<HederaFunctionality, List<Pair<DeterministicThrottle, Integer>>> reqLists =
+                new EnumMap<>(HederaFunctionality.class);
+
+        for (var bucket : defs.throttleBuckets()) {
+            try {
+                final var utilThrottleBucket = new ThrottleBucket<>(
+                        bucket.burstPeriodMs(),
+                        bucket.name(),
+                        bucket.throttleGroups().stream()
+                                .map(this::hapiGroupFromPbj)
+                                .toList());
+                var mapping = utilThrottleBucket.asThrottleMapping(CAPACITY_SPLIT);
+                var throttle = mapping.getLeft();
+                var reqs = mapping.getRight();
+                for (var req : reqs) {
+                    reqLists.computeIfAbsent(req.getLeft(), ignore -> new ArrayList<>())
+                            .add(Pair.of(throttle, req.getRight()));
+                }
+                newActiveThrottles.add(throttle);
+            } catch (IllegalStateException badBucket) {
+                log.error("When constructing bucket '{}' from state: {}", bucket.name(), badBucket.getMessage());
+            }
+        }
+        EnumMap<HederaFunctionality, ThrottleReqsManager> newFunctionReqs = new EnumMap<>(HederaFunctionality.class);
+        reqLists.forEach((function, reqs) -> newFunctionReqs.put(function, new ThrottleReqsManager(reqs)));
+
+        functionReqs = newFunctionReqs;
+        activeThrottles = newActiveThrottles;
+
+        logResolvedDefinitions(CAPACITY_SPLIT);
+    }
+
+    public void applyGasConfig() {
+        long capacity;
+
+        final var configuration = configProvider.getConfiguration();
+        final var contractsConfig = configuration.getConfigData(ContractsConfig.class);
+        if (contractsConfig.throttleThrottleByGas() && contractsConfig.maxGasPerSec() == 0) {
+            log.warn("Consensus gas throttling enabled, but limited to 0 gas/sec");
+            return;
+        } else {
```

Comment:
I'd always try to reduce the scope of a local variable to where it is used. For example, here, `capacity` is not used (or even initialized) until this line. You could rewrite it like this:
```
if (contractsConfig.throttleThrottleByGas() && contractsConfig.maxGasPerSec() == 0) {
    log.warn("Consensus gas throttling enabled, but limited to 0 gas/sec");
    return;
}
final long capacity = contractsConfig.maxGasPerSec();
```

It's not mandatory. The current code works just fine.

Reply:
Updated.